README.md

```diff
@@ -1,4 +1,231 @@
 [![Code Quality](https://img.shields.io/scrutinizer/g/bleech/acf-field-group-composer.svg?style=flat-square)](https://scrutinizer-ci.com/g/bleech/acf-field-group-composer)
 [![Code Coverage](https://img.shields.io/coveralls/bleech/acf-field-group-composer.svg?style=flat-square)](https://coveralls.io/github/bleech/acf-field-group-composer)
+
+> Configuration builder for advanced custom fields field groups
+
+This plugin helps you create reusable ACF fields and field groups with code.
+
+## Table of Contents
+
+- [Background](#background)
+- [Install](#install)
+- [Usage](#usage)
+- [API](#api)
+- [Maintainers](#maintainers)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Background
+
+ACF provides a great interface to create custom meta boxes in WordPress. However, there are to major downsides using it out of the box.
+
+First, configuring it via its GUI can be cumbersome if you have a lot of custom fields. Also fields created via the GUI are only stored in the database. That means migrating them between different stages of a development setup can only be done by migrating the database as well.
+
+Second, the local JSON feature, or `add_local_field_group`, is handy for checking the config into SCM and deploying it to different stages but it lacks some customizability, like reusing already specified fields in different context.
+
+acf-field-group-composer helps circumventing these downsides. It let's you use the build in `add_local_field_group` function, automatically adds unique keys to all fields added, and, if a field is not an array, but a string, it will apply a filter with that name and use the return value as a new field in the config.
+
+## Install
+
+TODO: install via WordPress instructions
+
+To install via composer, run
+
+```bash
+composer require flyntwp/acf-field-group-composer
+```
+
+## Usage
+
+To register a field group run
+
+```php
+ACFComposer\ACFComposer::registerFieldGroup($config);
+```
+
+The `$config` variable should be of the same format as the one you would pass to `acf_add_local_field_group`. There are only two things that are different:
+1. You do not have to specify a `key` in any field or group.
+2. A field group needs a unique `name` property.
+
+Following the minimal example [from ACF](https://www.advancedcustomfields.com/resources/register-fields-via-php)
+
+```php
+$config = [
+  'name' => 'group_1',
+  'title' => 'My Group',
+  'fields' => [
+    [
+      'label' => 'Sub Title',
+      'name' => 'sub_title',
+      'type' => 'text',
+    ]
+  ],
+  'location' => [
+    [
+      [
+      'param' => 'post_type',
+      'operator' => '==',
+      'value' => 'post',
+      ],
+    ],
+  ],
+];
+```
+
+In order to make use of the additional functionality of **acf-field-group-composer** you can extract the specified field and return in a filter. The name of the filter can be anything, however, we recommend a meaningful naming scheme like in the following example.
```

Comment:
In order to make use of the additional functionality of **acf-field-group-composer** you can extract the specified field and  return **it** in a filter. The name of the filter can be anything**, but** we recommend a meaningful naming scheme. **For example:**